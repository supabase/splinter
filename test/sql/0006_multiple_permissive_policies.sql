begin;
    set local search_path = '';

  -- No issues
  select * from lint."0006_multiple_permissive_policies";

  create table public.blog(
    id int primary key
  );

  alter table public.blog enable row level security;

  create policy select_own_posts on public.blog
  for select
  using (true);

  create policy all_own_posts on public.blog
  for all
  using (true);

  select * from lint."0006_multiple_permissive_policies";

  -- Now we drop the permissive policies and make sure that the same
  -- policies marked as "restrictive" to not flag
  drop policy select_own_posts on public.blog;
  drop policy all_own_posts on public.blog;

  create policy select_own_posts on public.blog
  as restrictive
  for select
  using (true);

  create policy all_own_posts on public.blog
  as restrictive
  for all
  using (true);

  -- No issues, the policies are restrictive, which is fine
  select * from lint."0006_multiple_permissive_policies";

  

rollback;
