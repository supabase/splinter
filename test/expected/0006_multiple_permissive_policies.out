begin;
    set local search_path = '';
  -- No issues
  select * from lint."0006_multiple_permissive_policies";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create table public.blog(
    id int primary key
  );
  alter table public.blog enable row level security;
  create policy select_own_posts on public.blog
  for select
  using (true);
  create policy all_own_posts on public.blog
  for all
  using (true);
  select * from lint."0006_multiple_permissive_policies";
             name             | level |  facing  |  categories   |                                                                                                                         description                                                                                                                         |                                                                             detail                                                                             |                                           remediation                                            |                       metadata                        |                           cache_key                           
------------------------------+-------+----------+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------+-------------------------------------------------------+---------------------------------------------------------------
 multiple_permissive_policies | WARN  | EXTERNAL | {PERFORMANCE} | Detects if multiple permissive row level security policies are present on a table for the same \`role\` and \`action\` (e.g. insert). Multiple permissive policies are suboptimal for performance as each policy must be executed for every relevant query. | Table \`public.blog\` has multiple permissive policies for role \`anon\` for action \`SELECT\`. Policies include \`{all_own_posts,select_own_posts}\`          | https://supabase.com/docs/guides/database/database-linter?lint=0006_multiple_permissive_policies | {"name": "blog", "type": "table", "schema": "public"} | multiple_permissive_policies_public_blog_anon_SELECT
 multiple_permissive_policies | WARN  | EXTERNAL | {PERFORMANCE} | Detects if multiple permissive row level security policies are present on a table for the same \`role\` and \`action\` (e.g. insert). Multiple permissive policies are suboptimal for performance as each policy must be executed for every relevant query. | Table \`public.blog\` has multiple permissive policies for role \`authenticated\` for action \`SELECT\`. Policies include \`{all_own_posts,select_own_posts}\` | https://supabase.com/docs/guides/database/database-linter?lint=0006_multiple_permissive_policies | {"name": "blog", "type": "table", "schema": "public"} | multiple_permissive_policies_public_blog_authenticated_SELECT
(2 rows)

  -- Now we drop the permissive policies and make sure that the same
  -- policies marked as "restrictive" to not flag
  drop policy select_own_posts on public.blog;
  drop policy all_own_posts on public.blog;
  create policy select_own_posts on public.blog
  as restrictive
  for select
  using (true);
  create policy all_own_posts on public.blog
  as restrictive
  for all
  using (true);
  -- No issues, the policies are restrictive, which is fine
  select * from lint."0006_multiple_permissive_policies";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
